Library/CorrelationIdManager.ts

```diff
@@ -0,0 +1,93 @@
+import https = require('https');
+import http = require('http');
+import url = require('url');
+
+class CorrelationIdManager {
+    public static correlationIdPrefix: "cid-v1:";
+
+    private static pendingLookups: {[key: string]: Function[]} = {};
+    private static completedLookups: {[key: string]: string} = {};

```

Comment:
These are used to implement the requirement that we only make a request once per session, even if someone creates multiple AI clients / configs.